tests/test_opentime.cpp

```diff
@@ -38,6 +38,75 @@ main(int argc, char** argv)
         assertFalse(t1 != t3);
     });
 
+    tests.add_test("test_from_time_string", [] {
+         std::string time_string = "0:12:04";
+         auto t = otime::RationalTime(24 * (12 * 60 + 4), 24);
+         auto time_obj = otime::RationalTime::from_time_string(time_string, 24);
+         assertTrue(t.almost_equal(time_obj, 0.001));
```

Comment:
Could you add a test for negative values, and another to test an invalid input?

Reply:
I should add assertions that `assertTrue` for invalid time returns for all the error conditions that are tested for in the parser...

Reply:
added a test to raise a value error for a negative time code, and also pink elephants.